### PR TITLE
Pass FrontJsonFapi to CuratedContentAgent

### PR DESCRIPTION
**Pull request by ioannakok**
Part of: https://github.com/guardian/dotcom-rendering/issues/5567

## What does this change?
* Adds `FrontJsonFapi` as a dependency of the `CuratedContentAgent`. In future PR the agent will call it to fetch the curated content from the fronts S3 bucket.
* Handles the wiring (dependency injection) of `FrontJsonFapi` in `article` and `preview` apps.
* Moves `CuratedContentAgent` and `OnwardsPicker` to `common` from `article`. They will be used by `ArticleController`, `LiveblogController` (in article) and `InteractiveController` (in interactives). 

Co-authored-by: Ashleigh Carr <ashcorr20@gmail.com>
Co-authored-by: Pete Faulconbridge <pete.faulconbridge@guardian.co.uk>
Co-authored-by: Pete Faulconbridge <peter.faulconbridge@gmail.com>
Co-authored-by: Marjan Kalanaki <marjan.kalanaki@guardian.co.uk>

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

File: article/conf/application.conf [hunk 1]
```diff
@@ -16,6 +16,14 @@ akka {
       }
     }
   }
+
+  blocking-operations {
```
Comment (ioannakok) [inline article/conf/application.conf:20]: Here we used the same values as: 

https://github.com/guardian/frontend/blob/main/admin/conf/application.conf#L26
https://github.com/guardian/frontend/blob/main/facia/conf/application.conf#L27

**Comment by OllysCoding** [comment]
This is maybe a silly question, but given that interactives are rendered in `applications` rather than `article`, should the `CuratedContentAgent` not live in `common` so it can wired into both apps?

File: article/conf/application.conf (hunk 1 above)
Comment (OllysCoding) [reply article/conf/application.conf:20]: What does the code do, I gather it's to do with some kind of concurrency?

File: article/test/agents/CuratedContentAgentTest.scala [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    extends AnyFlatSpec
+    with Matchers
+    with SingleServerSuite
+    with BeforeAndAfterAll
+    with WithMaterializer
+    with WithTestWsClient
+    with WithTestApplicationContext
+    with WithTestContentApiClient
+    with WithTestFrontJsonFapi {
+  "CuratedContentAgentTest" should "initialise with trails being an empty Seq" in {
```
Comment (AshCorr) [inline article/test/agents/CuratedContentAgentTest.scala:18]: Do we want to include the tests @bryophyta wrote in this PR?

**Comment by ioannakok** [comment]
> This is maybe a silly question, but given that interactives are rendered in `applications` rather than `article`, should the `CuratedContentAgent` not live in `common` so it can wired into both apps?

This is a great question! `CuratedContentAgent` and `OnwardsPicker` should live in `common`, you are absolutely right!

File: article/conf/application.conf (hunk 1 above)
Comment (ioannakok) [reply article/conf/application.conf:20]: This creates a dedicated thread pool for handling thread pool operations off the main thread. Here is the documentation: https://doc.akka.io/docs/akka/current/typed/dispatchers.html

Comment (ioannakok) [reply article/conf/application.conf:20]: `facia` also has a thread pool of size 128. We checked the size of the facia instances and it's the same as `article`, so don't expect to see any issues with that.

File: common/app/services/dotcomrendering/OnwardsPicker.scala [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package services.dotcomponents
+package services.dotcomrendering
```
Comment (mxdvl) [inline common/app/services/dotcomrendering/OnwardsPicker.scala:1]: I’m a fan of this change, but I’m not sure it should be included in this PR.

There’s also a lot of other places we would need to rename:
- [LiveBlogController.scala#L185-L186](https://github.com/guardian/frontend/blob/499856f7fd490ab200fd988cc20b4b856927362f/article/app/controllers/LiveBlogController.scala#L185-L186)
- [ArticlePicker.scala#L72-L78](https://github.com/guardian/frontend/blob/499856f7fd490ab200fd988cc20b4b856927362f/article/app/services/dotcomponents/ArticlePicker.scala#L72-L78)
- [DotcomRenderingService.scala#L93-L112](https://github.com/guardian/frontend/blob/499856f7fd490ab200fd988cc20b4b856927362f/common/app/renderers/DotcomRenderingService.scala#L93-L112)
- [DotcomponentsOnwardsModels.scala#L6](https://github.com/guardian/frontend/blob/499856f7fd490ab200fd988cc20b4b856927362f/onward/app/models/dotcomponents/DotcomponentsOnwardsModels.scala#L6)

**Review by mxdvl — COMMENTED** [review]
Can we isolate the rename of `dotcomponents` to `dotcomrendering` into its own PR?

File: common/app/services/dotcomrendering/OnwardsPicker.scala (hunk 3 above)
Comment (ioannakok) [reply common/app/services/dotcomrendering/OnwardsPicker.scala:1]: Agreed! This makes sense! I'll raise it straight after this gets merged

**Review by OllysCoding — APPROVED** [review]
+1 Looks good (thanks for talking this through with me!)

**Review by AshCorr — APPROVED** [review]
LGTM!

File: common/app/services/dotcomrendering/OnwardsPicker.scala (hunk 3 above)
Comment (ioannakok) [reply common/app/services/dotcomrendering/OnwardsPicker.scala:1]: Change made in this PR: https://github.com/guardian/frontend/pull/25506

**Pull request merged**